sql/core/src/main/scala/org/apache/spark/sql/DataFrameWriter.scala

```diff
@@ -303,7 +316,32 @@ final class DataFrameWriter private[sql](df: DataFrame) {
       conn.close()
     }
 
-    JdbcUtils.saveTable(df, url, table, props)
+    JdbcUtils.saveTable(df, url, table, props, columnMapping)
+  }
+
+  /**
+   * (java-specific) version of jdbc method
+   */
+   def jdbc(url: String,
+           table: String,
+           connectionProperties: Properties,
+           columnMapping: java.util.Map[String, String]): Unit = {
+    // Convert java Map into immutable scala Map
+    var sColumnMapping: scala.collection.immutable.Map[String, String] = null
```

Comment:
I am not sure I got "I agree it does matter whether you support null but you need not."